laa_crime_application_store_app/services/v1/notification_service.py

```diff
@@ -0,0 +1,64 @@
+from uuid import UUID
+
+import httpx
+import structlog
+from fastapi import Request
+from sqlalchemy.orm import Session
+
+from laa_crime_application_store_app.models.subscriber_schema import Subscriber
+
+logger = structlog.getLogger(__name__)
+
+
+class NotificationService:
+    @staticmethod
+    def subscribe(db: Session, subscriber_type: str, webhook_url: str):
+        existing_subscriber = (
+            db.query(Subscriber).filter(Subscriber.webhook_url == webhook_url).first()
+        )
+        if existing_subscriber is not None:
+            return None
+
+        new_subscriber = Subscriber(
+            subscriber_type=subscriber_type,
+            webhook_url=webhook_url,
+        )
+        db.add(new_subscriber)
+        db.commit()
+
+        return new_subscriber
+
+    @staticmethod
+    def unsubscribe(db: Session, subscriber_type: str, webhook_url: str):
+        existing_subscriber = (
+            db.query(Subscriber)
+            .filter(
+                Subscriber.webhook_url == webhook_url,
+                Subscriber.subscriber_type == subscriber_type,
+            )
+            .first()
```

Comment:
everyone keeps using first - it worries me that we never check to see how many rows filter returns but I guess if we never allow the same combination to be created then we should be ok.

Reply:
I think your composite primary key suggestion will fix this in this instance, as the database will guarantee there will never be more than one record that matches the criteria.